app/src/main/java/com/detroitlabs/chelseybaker/wundergroundweather/activities/LandingActivity.java

```
@@ -40,9 +47,44 @@ protected void onCreate(@Nullable final Bundle savedInstanceState) {
         if (searchListLocationButton != null) {
             searchListLocationButton.setOnClickListener(this);
         }
+
+        final Button searchByLocationButton = (Button)findViewById(R.id.search_current_location_button);
+        if (searchByLocationButton != null) {
+            searchByLocationButton.setOnClickListener(this);
+
+            // Create an instance of GoogleAPIClient.
+            if (mGoogleApiClient == null) {
+                mGoogleApiClient = new GoogleApiClient.Builder(this)
+                        .addConnectionCallbacks(this)
+                        .addOnConnectionFailedListener(this)
+                        .addApi(LocationServices.API)
+                        .build();
+
+
+            }
+        }
     }
 
     @Override
+    protected void onStart() {
+        if (mGoogleApiClient != null) {
```

Should be safe to skip this since `onCreate` is guaranteed to be called before `onStart`. While it's generally good to catch nulls in places where they should never show, you also want to balance that with having the app fail loudly during development (so you can fix programmer errors). Here's a case where I'd probably prefer a loud failure, as otherwise I may sneak code into production that's never actually executed!